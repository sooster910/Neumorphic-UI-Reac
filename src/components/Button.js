/* eslint-disable */
import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';

const StyledButton = styled.button`
  min-width: 30px;
  width: auto;
  padding: 0 16px;
  box-sizing: border-box;
  display: inline-flex;
  justify-content: center;
  align-items: center;
  padding: 5px 10px;
  height: 44px;
  border-radius: 1rem;
  border-style: none;
  outline: none;
  box-shadow: ${props =>
    props.dark
      ? '5px 5px 12px #2b2b2b, -4px -4px 10px #3b3b3b'
      : '2px 2px 3px #c8d0e7, -5px -5px 5px #eef3f9'};
  transition: box-shadow 0.2s ease;
  background: ${props => (props.dark ? '#333333' : '#e4ebf5')};
  color: #f2994a;
  outline: none;

  :hover {
    color: #4b4b96;
    outline: none;
  }
  :active {
    box-shadow: ${props =>
      props.dark
        ? 'inset -2px -2px 5px #4b4b4b,inset 2px 2px 5px #212121'
        : 'inset -2px -2px 1px #ffffff,inset 2px 2px 5px #cdd3dc'};
    outline: none;
  }
`;

const Button = ({ title,...rest }) => {
  return <StyledButton {...rest}>{title}</StyledButton>;
};

Button.propTypes = {
  title: propTypes.string,
};

Button.defaultProps = {
  title: 'Button',
};

export default Button;
