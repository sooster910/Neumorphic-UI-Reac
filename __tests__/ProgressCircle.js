

/**
 *  1. METHOD :  USING MOUNT
 */
import React from 'react';
import { mount } from 'enzyme';

import  Button from '../src/components/Button';
import 'jest-styled-components';

describe('<Button />', () => {
  it('matches snapshot', () => {
    const wrapper = mount(<Button title="Hi"/>);
    expect(wrapper).toMatchSnapshot();
  });
});

/**
 *  2. METHOD :  USING TREE RENDERER 
 */

// import React from 'react'
// import renderer from 'react-test-renderer'
// import Button from '../src/components/Button'
// import 'jest-styled-components';


// describe('Button Block', () => {
//   it('should render correctly', () => {
//     const tree = renderer
//     .create(<Button>Submit</Button>)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
// });