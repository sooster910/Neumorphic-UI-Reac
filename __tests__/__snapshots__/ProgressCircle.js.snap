// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
.c0 {
  min-width: 30px;
  width: auto;
  padding: 0 16px;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5px 10px;
  height: 44px;
  border-radius: 1rem;
  border-style: none;
  outline: none;
  box-shadow: 2px 2px 3px #c8d0e7,-5px -5px 5px #eef3f9;
  -webkit-transition: box-shadow 0.2s ease;
  transition: box-shadow 0.2s ease;
  background: #e4ebf5;
  color: #f2994a;
  outline: none;
}

.c0:hover {
  color: #4b4b96;
  outline: none;
}

.c0:active {
  box-shadow: inset -2px -2px 1px #ffffff,inset 2px 2px 5px #cdd3dc;
  outline: none;
}

<Button
  title="Hi"
>
  <styled.button>
    <button
      className="c0"
    >
      Hi
    </button>
  </styled.button>
</Button>
`;
